components/tensorflow-notebook-image/Dockerfile

```diff
@@ -93,55 +93,24 @@ RUN cd /tmp && \
     mv tini /usr/local/bin/tini && \
     chmod +x /usr/local/bin/tini
 
-# Install conda as jovyan user and check the md5 sum provided on the download site
```

Comment:
Thanks @jlewi. I wanted to be sure that there are no issues if conda is introduced by custom images.

Reply:
Would it then make sense to list some of the kubeflow specific installs in the docs which would make it more seamless to work with. eg. pipelines SDK etc. Or show the dockerfile that you are providing by default (as a template) so that the community could tailor it for their own needs. Adding other kernels and what not.

Reply:
@leonardaukea what does a kubeflow specific install mean in this context?

In terms of documentation how could the doc https://www.kubeflow.org/docs/notebooks/custom-notebook/ be improved? I think there is already instructions there about how to write your Dockerfile to build custom images

Reply:
@jlewi we are currently following the approach of custom jupyter images with conda and additional python libraries both external and internal. For internal libraries and few external libraries, users wanted the flexibility to install newer packages without dependency on middleware team to provide new docker images. So the custom image building is very helpful.

What exactly is packaged in community offered notebook image can now be understood only by looking into the docker file in git. Instead, if it's made available in documentation site, it will be easier to end users. Of course there is some Dev overhead.

Reply:
@jlewi Initially,  I was trying to say that it would be helpful to show the community Dockerfile in the docs or, some light-weight version of it. The "kubeflow specific Installs" was referring to packages and settings that are used, apart from the minimal requirements that are provided in the docs. 

Secondly, I think the docs needs to be changed so to add a description for jupyter lab which I think is more popular now (It should be). I had to make minor changes in order for it to work ( `NotebookApp`-> `LabApp`). 

Also, I'm assuming that the community image includes things like `gsutil` since all of the examples are very "Gcloud".  Maybe it makes sense to provide a template image for each cloud provider. Or at least provide suggestions. 

It  would also be helpful to describe how you can add your custom image to the list of available images in the dropdown in the UI.  By editing the `jupyter-web-app-config`.

Btw, is it not a security risk to use `--allow-root`?  Why is this a requirement?

Reply:
@LeonardAukea @nrchakradhar Can you file issues about updating the docs?

@LeonardAukea running as root is a security risk and I don't think we should run as root. But I believe that is actually determined by the notebook controller and the value of USER which will determine what user the container starts with.